src/cargo/ops/cargo_output_metadata.rs

```diff
@@ -0,0 +1,146 @@
+use std::collections::HashMap;
+use std::io::Write;
+use std::fs;
+use std::path::{Path, PathBuf};
+use rustc_serialize::{json, Decoder, Decodable};
+use toml;
+use core::resolver::Resolve;
+use core::{Source, Package};
+use sources::PathSource;
+use util::CargoResult;
+use util::config::Config;
+use ops;
+
+
+#[derive(RustcDecodable)]
```

Comment:
Huh I'm kinda surprised this works, does this emit strings by default? It may be better to explicitly just take a string and then convert it to an enum manually (may give better errors as well)